### ISSUE TITLE
Array query params

### ISSUE THREAD

**Issue by mmcc** [rhazdon/blue_bird#34]
I'm going to preface this by saying I don't think this is [supported by API Blueprint right now](https://github.com/apiaryio/api-blueprint/issues/240), but I figured it could be useful as work continues on Swagger support.

Currently, the support for array params is pretty broken. Our query params that are arrays follow the standard Phoenix syntax, `foo[]=1&foo[]=21`. As far as I can tell, there's currently no way to describe this in the `api` macro (but again, I think that's a problem all the way down to API Blueprint). 

We've ended up going with just using a string for the name and setting it to `param[]` so it at least shows up correctly in the URI Parameter list, but unfortunately, this means that the params won't get picked up by the `ConnLogger`. Not a huge deal on our end, I think this solution gets us close enough, but it'd be cool to find a solution for this long term.

**Comment by woylie** [comment]
Here is a more recent [proposal](https://github.com/apiaryio/api-blueprint-rfcs/pull/3) to add support to the Blueprint format. Swagger on the other hand supports array params. We should define a syntax for defining them in BlueBird and figure out a workaround for the apib output.

**Comment by mmcc** [comment]
Yeah, I think I saw that proposal when I was digging into it but seems like that one's pretty dead on the vine. I agree, feels like step one is just to make this work well with Swagger output and then work backward to a reasonable apib solution if/until the proposal makes it in.

**Comment by gonzag88** [comment]
@mmcc Could you post an example of how you've ended up going using a string for the name?

Thanks!

**Comment by woylie** [comment]
@gonzag88 I think he just did something like this:

```
parameter "foo[]", :string, [description: "some description"]
```

**Comment by mmcc** [comment]
@gonzag88 Yep, @woylie hit the nail on the head, but as I mentioned that was a pretty half-baked solution. I think what we've _actually_ ended up doing is using the param without the brackets and then using good ol' `sed` to search/replace in the html output of Aglio 😬

**Comment by gonzag88** [comment]
Thanks @mmcc!!
